### PR TITLE
chore: handle 1.11 mypy upgrades

### PR DESCRIPTION
**Pull request by antazoey**
### What I did

fixes new mypy complaints and upgrades

### Checklist

- [ ] All changes are completed
- [ ] New test cases have been added
- [ ] Documentation has been updated

### CODE REVIEW

File: src/ape/utils/basemodel.py [hunk 1]
```diff
@@ -201,7 +201,7 @@ def __getattr__(self, item):
     def __getitem__(self, item):
         return self._callback(item)
 
-    def get(self, item):
+    def get(self, item):  # type: ignore[override]
```
Comment (antazoey) [inline src/ape/utils/basemodel.py:204]: basically consider this an interface and we are never using any other kwargs in the context it gets called

File: src/ape_run/_cli.py [hunk 2]
```diff
@@ -213,7 +213,7 @@ def get_command(self, ctx, name):
 
         # NOTE: don't return anything so Click displays proper error
 
-    def result_callback(self, result, interactive):
+    def result_callback(self, result, interactive: bool):  # type: ignore[override]
```
Comment (antazoey) [inline src/ape_run/_cli.py:216]: we just be hacking click a bit here, dont mind us, mypy

Comment (fubuloubu) [reply src/ape_run/_cli.py:216]: the hack is annotating `interactive`?

Comment (antazoey) [reply src/ape_run/_cli.py:216]: the hack is having `interactive` at all, it isnt in the original method. There are no real changes here.

**Review by fubuloubu — APPROVED** [review]

**Pull request merged**